h2/src/main/org/h2/mvstore/DataUtils.java

```diff
@@ -726,7 +726,7 @@ public static String getMapName(String s) {
      */
     public static int getFletcher32(byte[] bytes, int length) {
         int s1 = 0xffff, s2 = 0xffff;
-        int i = 0, evenLength = length / 2 * 2;
+        int i = 0, evenLength = length & ~1;
```

Comment:
this kind of trick is unnecessary, the compiler will always do a better job of picking instructions at this level, and clarity is more important

Reply:
`/2*2` is not very readable and bit mask is also a common practice for taking the even part.

I'll check machine code on x86-64 later.

Reply:
Checked on Java 8 and Java 9 on x86-64 Server VM.

JIT compiler optimizes `* 2` to `<< 1`, but performs `/ 2` with check for `-1` (special case on x86 that Java handles in own way) and `idiv` instruction.

Please also notice that `/ 2 * 2` is not the same as `& -1`, both return the same results for non-negative integers (its our case), but different for negative so it probably will not be optimized it this way in future versions of JIT compilers too.